### PR TITLE
Fix: Eye Dropper not working on certain panel areas [ED-3411]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: modules/elements-color-picker/assets/js/editor/commands/enter-preview.js [hunk 1]
```diff
@@ -2,8 +2,22 @@ import CommandBase from 'elementor-api/modules/command-base';
...
-		this.component.currentPicker.container.view.renderUI();
+		container.settings.set( control, args.value );
+
+		const { view } = container;
+
+		if ( view?.renderUI ) {
+			view.renderUI();
+		}
+
+		if ( kit ) {
```
Comment (matipojo) [inline modules/elements-color-picker/assets/js/editor/commands/enter-preview.js:16]: An empty object is `true`, see assets/dev/js/editor/controls/color.js:90

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/elements-color-picker/assets/js/editor/commands/enter-preview.js (hunk 1 above)
Comment (StyleShit) [reply modules/elements-color-picker/assets/js/editor/commands/enter-preview.js:16]: Oops, meant to write null..

File: assets/dev/js/editor/controls/color.js [hunk 2]
```diff
@@ -84,12 +84,20 @@ export default class extends ControlBaseDataView {
...
+
+		// When it's a kit (i.e "Site Settings").
+		if ( 'kit' === container.document.config.type ) {
+			kit = container.document;
+		}
 
 		$colorPicker.on( 'click', () => {
 			$e.run( 'elements-color-picker/start', {
-				container: currentElement.container,
+				container,
```
Comment (iNewLegend) [inline assets/dev/js/editor/controls/color.js:99]: A bit confusing... can you add JSDOC for `this.component.currentPicker`, to specify object properties?

File: modules/elements-color-picker/assets/js/editor/commands/enter-preview.js [hunk 3]
```diff
@@ -2,8 +2,22 @@ import CommandBase from 'elementor-api/modules/command-base';
...
+			const { id } = kit.config,
+				cssVar = `--e-global-color-${ container.id }`;
+
+			elementor.$previewContents[ 0 ].querySelector( `.elementor-kit-${ id }` ).style.setProperty( cssVar, args.value );
+		}
```
Comment (iNewLegend) [inline modules/elements-color-picker/assets/js/editor/commands/enter-preview.js:21]:
```suggestion
			const { id } = kit.config,
				cssVar = `--e-global-color-${ container.id }`;

			elementor.$previewContents[ 0 ].querySelector( `.elementor-kit-${ id }` ).style.setProperty( cssVar, args.value );
		}
```

you do it twice, you can have a function in component to handle it.

**Review by iNewLegend — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/controls/color.js (hunk 2 above)
Comment (StyleShit) [reply assets/dev/js/editor/controls/color.js:99]: Good idea :)

File: modules/elements-color-picker/assets/js/editor/commands/enter-preview.js (hunk 3 above)
Comment (StyleShit) [reply modules/elements-color-picker/assets/js/editor/commands/enter-preview.js:21]: Yup, kinda stupid that I haven't done that at the first time.

**Review by iNewLegend — APPROVED** [review]

File: modules/elements-color-picker/assets/js/editor/commands/enter-preview.js [hunk 4]
```diff
@@ -2,8 +2,7 @@ import CommandBase from 'elementor-api/modules/command-base';
 
 export class EnterPreview extends CommandBase {
 	apply( args ) {
-		// Silent
-		this.component.currentPicker.container.settings.set( this.component.currentPicker.control, args.value );
-		this.component.currentPicker.container.view.renderUI();
+		// Silent.
```
Comment (matipojo) [inline modules/elements-color-picker/assets/js/editor/commands/enter-preview.js:5]:
```suggestion
```

File: modules/elements-color-picker/assets/js/editor/commands/exit-preview.js [hunk 5]
```diff
@@ -2,12 +2,13 @@ import CommandBase from 'elementor-api/modules/command-base';
...
+		const { initialColor } = this.component.currentPicker;
+
+		if ( null === initialColor ) {
 			return;
 		}
 
-		// Silent
-		this.component.currentPicker.container.settings.set( this.component.currentPicker.control, this.component.currentPicker.initialColor );
-		this.component.currentPicker.container.view.renderUI();
+		// Silent.
```
Comment (matipojo) [inline modules/elements-color-picker/assets/js/editor/commands/exit-preview.js:11]:
```suggestion
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**